### ISSUE TITLE
Basic HTTP Authentication Support

### ISSUE THREAD

**Issue by stirno** [stirno/FluentAutomation#58]
Allow automation of sites protected via Basic HTTP Authentication.

**Comment by detroitpro** [comment]
Has there been any traction on this issue?

**Comment by stirno** [comment]
Ya, I had a few different paths to make this work but they weren't reliable.

For some sites, the easiest way is to put the user:pass in the URL (the common advice for people using WebDriver):

https://user:pass@domain.tld/

This bypasses the dialog entirely in most browsers. I want the ultimate fix to be more transparent and not rely on this behavior and we haven't had time to get that moving. It will happen at some point, either due to WebDriver adding better support that we can then tap into or via our own implementation of a passthrough proxy for the test but that brings on other problems..

TLDR: yes, but we have no elegant solution yet so we haven't pushed anything. Use the URL trick with I.Open() and it should get you through any auth dialogs in most browsers.

**Comment by stirno** [comment]
Despite many attempts we can't make this work in a way that is consistent across all the browsers we support without the URL trick... and anything we do would likely be a hack to 'force' that into the URL which is undesirable 'magic' behavior. Leaving it to the user unless theres a big advance in how Selenium handles it at least. :(
